File: skills/opencode-api/opencode-api.js
/**
 * opencode-api.js - opencode REST API JavaScript 客户端
 * 
 * 用法:
 * const client = new OpenCodeAPI({
 *   baseURL: process.env.OPENCODE_SERVER_BASE_URL || 'http://127.0.0.1:4096',
 *   username: process.env.OPENCODE_SERVER_USERNAME || 'opencode',
 *   password: process.env.OPENCODE_SERVER_PASSWORD // 可选
 * });
 */

class OpenCodeAPI {
  constructor(options = {}) {
    this.baseURL = options.baseURL || process.env.OPENCODE_SERVER_BASE_URL || 'http://127.0.0.1:4096';
    this.username = options.username || process.env.OPENCODE_SERVER_USERNAME || 'opencode';
    this.password = options.password || process.env.OPENCODE_SERVER_PASSWORD || '';
    this.authHeader = this.password 
      ? 'Basic ' + Buffer.from(`${this.username}:${this.password}`).toString('base64')
      : null;
  }

  async request(method, path, body = null) {
    const url = `${this.baseURL}${path}`;
    const headers = {
      'Content-Type': 'application/json',
    };
    
    if (this.authHeader) {
      headers['Authorization'] = this.authHeader;
    }

    const options = {
      method,
      headers,
    };

    if (body && (method === 'POST' || method === 'PATCH' || method === 'PUT')) {
      options.body = JSON.stringify(body);
    }

    const response = await fetch(url, options);
    
    if (!response.ok) {
      const error = await response.text();
      throw new Error(`API Error ${response.status}: ${error}`);
    }

    // 处理空响应
    if (response.status === 204) {
      return null;
    }

    return response.json();
  }

  // ========== 全局端点 ==========
  
  async health() {
    return this.request('GET', '/global/health');
  }

  async getGlobalConfig() {
    return this.request('GET', '/global/config');
  }

  async updateGlobalConfig(config) {
    return this.request('PATCH', '/global/config', config);
  }

  async dispose() {
    return this.request('POST', '/global/dispose');
  }

  // ========== 会话管理 ==========
  
  async listSessions() {
    return this.request('GET', '/session');
  }

  async createSession(options = {}) {
    const { parentID, title, directory, workspace } = options;
    const body = {};
    if (parentID) body.parentID = parentID;
    if (title) body.title = title;
    return this.request('POST', '/session', body);
  }

  async getSession(sessionID) {
    return this.request('GET', `/session/${sessionID}`);
  }

  async deleteSession(sessionID) {
    return this.request('DELETE', `/session/${sessionID}`);
  }

  async updateSession(sessionID, updates) {
    return this.request('PATCH', `/session/${sessionID}`, updates);
  }

  async abortSession(sessionID) {
    return this.request('POST', `/session/${sessionID}/abort`);
  }

  async forkSession(sessionID, messageID = null) {
    const body = messageID ? { messageID } : {};
    return this.request('POST', `/session/${sessionID}/fork`, body);
  }

  async summarizeSession(sessionID, options) {
    return this.request('POST', `/session/${sessionID}/summarize`, options);
  }

  async shareSession(sessionID) {
    return this.request('POST', `/session/${sessionID}/share`);
  }

  async unshareSession(sessionID) {
    return this.request('DELETE', `/session/${sessionID}/share`);
  }

  async getSessionDiff(sessionID, messageID = null) {
    const path = messageID 
      ? `/session/${sessionID}/diff?messageID=${messageID}`
      : `/session/${sessionID}/diff`;
    return this.request('GET', path);
  }

  async getSessionTodo(sessionID) {
    return this.request('GET', `/session/${sessionID}/todo`);
  }

  async getSessionChildren(sessionID) {
    return this.request('GET', `/session/${sessionID}/children`);
  }

  async getSessionStatus() {
    return this.request('GET', '/session/status');
  }

  // ========== 消息 ==========
  
  async listMessages(sessionID, options = {}) {
    const { limit, before } = options;
    const params = new URLSearchParams();
    if (limit) params.append('limit', limit);
    if (before) params.append('before', before);
    const query = params.toString() ? `?${params}` : '';
    return this.request('GET', `/session/${sessionID}/message${query}`);
  }

  async getMessage(sessionID, messageID) {
    return this.request('GET', `/session/${sessionID}/message/${messageID}`);
  }

  async sendMessage(sessionID, options) {
    const {
      messageID,
      model,
      agent,
      noReply,
      system,
      variant,
      parts,
      format,
    } = options;

    const body = { parts };
    if (messageID) body.messageID = messageID;
    if (model) body.model = model;
    if (agent) body.agent = agent;
    if (noReply !== undefined) body.noReply = noReply;
    if (system) body.system = system;
    if (variant) body.variant = variant;
    if (format) body.format = format;

    return this.request('POST', `/session/${sessionID}/message`, body);
  }

  async sendPromptAsync(sessionID, options) {
    // 异步发送，不等待响应
    const body = { ...options };
    return this.request('POST', `/session/${sessionID}/prompt_async`, body);
  }

  async deleteMessage(sessionID, messageID) {
    return this.request('DELETE', `/session/${sessionID}/message/${messageID}`);
  }

  async revertMessage(sessionID, messageID, partID = null) {
    const body = partID ? { messageID, partID } : { messageID };
    return this.request('POST', `/session/${sessionID}/revert`, body);
  }

  async unrevertMessages(sessionID) {
    return this.request('POST', `/session/${sessionID}/unrevert`);
  }

  // ========== 命令 ==========
  
  async executeCommand(sessionID, options) {
    const { command, arguments: args, agent, model, messageID } = options;
    const body = { command, arguments: args };
    if (agent) body.agent = agent;
    if (model) body.model = model;
    if (messageID) body.messageID = messageID;
    return this.request('POST', `/session/${sessionID}/command`, body);
  }

  async runShell(sessionID, options) {
    const { command, agent, model } = options;
    const body = { command, agent };
    if (model) body.model = model;
    return this.request('POST', `/session/${sessionID}/shell`, body);
  }

  // ========== 项目 ==========
  
  async listProjects(options = {}) {
    const { directory, workspace } = options;
    const params = new URLSearchParams();
    if (directory) params.append('directory', directory);
    if (workspace) params.append('workspace', workspace);
    const query = params.toString() ? `?${params}` : '';
    return this.request('GET', `/project${query}`);
  }

  async getCurrentProject(options = {}) {
    const { directory, workspace } = options;
    const params = new URLSearchParams();
    if (directory) params.append('directory', directory);
    if (workspace) params.append('workspace', workspace);
    const query = params.toString() ? `?${params}` : '';
    return this.request('GET', `/project/current${query}`);
  }

  async initGit(options = {}) {
    const { directory, workspace } = options;
    const params = new URLSearchParams();
    if (directory) params.append('directory', directory);
    if (workspace) params.append('workspace', workspace);
    const query = params.toString() ? `?${params}` : '';
    return this.request('POST', `/project/git/init${query}`);
  }

  async updateProject(projectID, updates) {
    return this.request('PATCH', `/project/${projectID}`, updates);
  }

  // ========== 文件 ==========
  
  async listFiles(path = '.', options = {}) {
    const { directory, workspace } = options;
    const params = new URLSearchParams({ path });
    if (directory) params.append('directory', directory);
    if (workspace) params.append('workspace', workspace);
    return this.request('GET', `/file?${params}`);
  }

  async readFile(path, options = {}) {
    const { directory, workspace } = options;
    const params = new URLSearchParams({ path });
    if (directory) params.append('directory', directory);
    if (workspace) params.append('workspace', workspace);
    return this.request('GET', `/file/content?${params}`);
  }

  async getFileStatus() {
    return this.request('GET', '/file/status');
  }

  // ========== 搜索 ==========
  
  async searchText(pattern, options = {}) {
    const { directory, workspace } = options;
    const params = new URLSearchParams({ pattern });
    if (directory) params.append('directory', directory);
    if (workspace) params.append('workspace', workspace);
    return this.request('GET', `/find?${params}`);
  }

  async findFiles(query, options = {}) {
    const { type, directory, limit, dirs } = options;
    const params = new URLSearchParams({ query });
    if (type) params.append('type', type);
    if (directory) params.append('directory', directory);
    if (limit) params.append('limit', limit);
    if (dirs !== undefined) params.append('dirs', String(dirs));
    return this.request('GET', `/find/file?${params}`);
  }

  async findSymbols(query) {
    return this.request('GET', `/find/symbol?query=${encodeURIComponent(query)}`);
  }

  // ========== 配置 ==========
  
  async getConfig() {
    return this.request('GET', '/config');
  }

  async updateConfig(updates) {
    return this.request('PATCH', '/config', updates);
  }

  async getProviders() {
    return this.request('GET', '/config/providers');
  }

  async listProviders() {
    return this.request('GET', '/provider');
  }

  async getProviderAuth() {
    return this.request('GET', '/provider/auth');
  }

  async setAuth(providerID, credentials) {
    return this.request('PUT', `/auth/${providerID}`, credentials);
  }

  async removeAuth(providerID) {
    return this.request('DELETE', `/auth/${providerID}`);
  }

  // ========== 代理/命令/技能 ==========
  
  async listAgents() {
    return this.request('GET', '/agent');
  }

  async listCommands() {
    return this.request('GET', '/command');
  }

  async listSkills() {
    return this.request('GET', '/skill');
  }

  // ========== MCP ==========
  
  async getMCPStatus() {
    return this.request('GET', '/mcp');
  }

  async addMCPServer(name, config) {
    return this.request('POST', '/mcp', { name, config });
  }

  async connectMCPServer(name) {
    return this.request('POST', `/mcp/${name}/connect`);
  }

  async disconnectMCPServer(name) {
    return this.request('POST', `/mcp/${name}/disconnect`);
  }

  // ========== 其他 ==========
  
  async getVCSInfo() {
    return this.request('GET', '/vcs');
  }

  async getPathInfo() {
    return this.request('GET', '/path');
  }

  async getLSPStatus() {
    return this.request('GET', '/lsp');
  }

  async getFormatterStatus() {
    return this.request('GET', '/formatter');
  }

  async listPermissions() {
    return this.request('GET', '/permission');
  }

  async respondToPermission(sessionID, permissionID, response, remember = false) {
    return this.request('POST', `/session/${sessionID}/permissions/${permissionID}`, {
      response,
      remember,
    });
  }

  async listPTY() {
    return this.request('GET', '/pty');
  }

  async createPTY() {
    return this.request('POST', '/pty');
  }
}

// 导出
if (typeof module !== 'undefined' && module.exports) {
  module.exports = { OpenCodeAPI };
}

// 使用示例
/*
const client = new OpenCodeAPI({
  baseURL: process.env.OPENCODE_SERVER_BASE_URL || 'http://127.0.0.1:4096',
  username: process.env.OPENCODE_SERVER_USERNAME || 'opencode',
  password: process.env.OPENCODE_SERVER_PASSWORD
});

// 健康检查
const health = await client.health();
console.log(health);

// 创建会话
const session = await client.createSession({ title: '我的会话' });
console.log('Session ID:', session.id);

// 发送消息
const response = await client.sendMessage(session.id, {
  parts: [{ type: 'text', text: '你好！' }]
});
console.log(response);

// 列出文件
const files = await client.listFiles('.');
console.log(files);
*/

File: skills/opencode-api/opencode-api.py
"""
opencode_api.py - opencode REST API Python 客户端

用法:
    import os
    from opencode_api import OpenCodeAPI
    
    client = OpenCodeAPI(
        base_url=os.getenv('OPENCODE_SERVER_BASE_URL', 'http://127.0.0.1:4096'),
        username=os.getenv('OPENCODE_SERVER_USERNAME', 'opencode'),
        password=os.getenv('OPENCODE_SERVER_PASSWORD')  # 可选
    )
    
    # 健康检查
    health = client.health()
    
    # 创建会话
    session = client.create_session(title='我的会话')
    
    # 发送消息
    response = client.send_message(session['id'], '你好！')
"""

import requests
import os
from typing import Optional, Dict, List, Any
from urllib.parse import urlencode


class OpenCodeAPI:
    def __init__(
        self,
        base_url: str = None,
        username: str = None,
        password: Optional[str] = None,
    ):
        self.base_url = (base_url or os.getenv('OPENCODE_SERVER_BASE_URL', 'http://127.0.0.1:4096')).rstrip('/')
        self.username = username or os.getenv('OPENCODE_SERVER_USERNAME', 'opencode')
        self.password = password or os.getenv('OPENCODE_SERVER_PASSWORD')
        self.base_url = base_url.rstrip('/')
        self.username = username
        self.password = password
        self.session = requests.Session()
        
        if password:
            self.session.auth = (username, password)
        
        self.session.headers.update({'Content-Type': 'application/json'})

    def _request(self, method: str, path: str, json: Optional[Dict] = None, params: Optional[Dict] = None) -> Any:
        url = f"{self.base_url}{path}"
        response = self.session.request(method, url, json=json, params=params)
        response.raise_for_status()
        
        if response.status_code == 204:
            return None
        
        return response.json()

    # ========== 全局端点 ==========
    
    def health(self) -> Dict:
        """健康检查"""
        return self._request('GET', '/global/health')

    def get_global_config(self) -> Dict:
        """获取全局配置"""
        return self._request('GET', '/global/config')

    def update_global_config(self, config: Dict) -> Dict:
        """更新全局配置"""
        return self._request('PATCH', '/global/config', json=config)

    def dispose(self) -> bool:
        """释放所有资源"""
        return self._request('POST', '/global/dispose')

    # ========== 会话管理 ==========
    
    def list_sessions(self) -> List[Dict]:
        """列出所有会话"""
        return self._request('GET', '/session')

    def create_session(self, parent_id: Optional[str] = None, title: Optional[str] = None) -> Dict:
        """创建新会话"""
        body = {}
        if parent_id:
            body['parentID'] = parent_id
        if title:
            body['title'] = title
        return self._request('POST', '/session', json=body)

    def get_session(self, session_id: str) -> Dict:
        """获取会话详情"""
        return self._request('GET', f'/session/{session_id}')

    def delete_session(self, session_id: str) -> bool:
        """删除会话"""
        return self._request('DELETE', f'/session/{session_id}')

    def update_session(self, session_id: str, updates: Dict) -> Dict:
        """更新会话"""
        return self._request('PATCH', f'/session/{session_id}', json=updates)

    def abort_session(self, session_id: str) -> bool:
        """中止会话"""
        return self._request('POST', f'/session/{session_id}/abort')

    def fork_session(self, session_id: str, message_id: Optional[str] = None) -> Dict:
        """分叉会话"""
        body = {'messageID': message_id} if message_id else {}
        return self._request('POST', f'/session/{session_id}/fork', json=body)

    def summarize_session(self, session_id: str, provider_id: str, model_id: str) -> bool:
        """总结会话"""
        return self._request('POST', f'/session/{session_id}/summarize', json={
            'providerID': provider_id,
            'modelID': model_id
        })

    def share_session(self, session_id: str) -> Dict:
        """分享会话"""
        return self._request('POST', f'/session/{session_id}/share')

    def unshare_session(self, session_id: str) -> Dict:
        """取消分享会话"""
        return self._request('DELETE', f'/session/{session_id}/share')

    def get_session_diff(self, session_id: str, message_id: Optional[str] = None) -> List[Dict]:
        """获取会话差异"""
        params = {'messageID': message_id} if message_id else {}
        return self._request('GET', f'/session/{session_id}/diff', params=params)

    def get_session_todo(self, session_id: str) -> List[Dict]:
        """获取待办事项"""
        return self._request('GET', f'/session/{session_id}/todo')

    def get_session_children(self, session_id: str) -> List[Dict]:
        """获取子会话"""
        return self._request('GET', f'/session/{session_id}/children')

    def get_session_status(self) -> Dict:
        """获取所有会话状态"""
        return self._request('GET', '/session/status')

    # ========== 消息 ==========
    
    def list_messages(
        self, 
        session_id: str, 
        limit: Optional[int] = None, 
        before: Optional[str] = None
    ) -> List[Dict]:
        """获取消息列表"""
        params = {}
        if limit:
            params['limit'] = limit
        if before:
            params['before'] = before
        return self._request('GET', f'/session/{session_id}/message', params=params)

    def get_message(self, session_id: str, message_id: str) -> Dict:
        """获取单条消息"""
        return self._request('GET', f'/session/{session_id}/message/{message_id}')

    def send_message(
        self,
        session_id: str,
        parts: List[Dict],
        model: Optional[Dict] = None,
        agent: Optional[str] = None,
        no_reply: Optional[bool] = None,
        system: Optional[str] = None,
        variant: Optional[str] = None,
        format: Optional[Dict] = None,
    ) -> Dict:
        """发送消息"""
        body = {'parts': parts}
        if model:
            body['model'] = model
        if agent:
            body['agent'] = agent
        if no_reply is not None:
            body['noReply'] = no_reply
        if system:
            body['system'] = system
        if variant:
            body['variant'] = variant
        if format:
            body['format'] = format
        return self._request('POST', f'/session/{session_id}/message', json=body)

    def send_prompt_async(self, session_id: str, options: Dict) -> None:
        """异步发送消息（不等待响应）"""
        return self._request('POST', f'/session/{session_id}/prompt_async', json=options)

    def delete_message(self, session_id: str, message_id: str) -> bool:
        """删除消息"""
        return self._request('DELETE', f'/session/{session_id}/message/{message_id}')

    def revert_message(
        self, 
        session_id: str, 
        message_id: str, 
        part_id: Optional[str] = None
    ) -> bool:
        """回退消息"""
        body = {'messageID': message_id, 'partID': part_id} if part_id else {'messageID': message_id}
        return self._request('POST', f'/session/{session_id}/revert', json=body)

    def unrevert_messages(self, session_id: str) -> bool:
        """恢复所有回退的消息"""
        return self._request('POST', f'/session/{session_id}/unrevert')

    # ========== 命令 ==========
    
    def execute_command(
        self,
        session_id: str,
        command: str,
        arguments: Optional[Dict] = None,
        agent: Optional[str] = None,
        model: Optional[Dict] = None,
    ) -> Dict:
        """执行斜杠命令"""
        body = {'command': command, 'arguments': arguments or {}}
        if agent:
            body['agent'] = agent
        if model:
            body['model'] = model
        return self._request('POST', f'/session/{session_id}/command', json=body)

    def run_shell(
        self,
        session_id: str,
        command: str,
        agent: str,
        model: Optional[Dict] = None,
    ) -> Dict:
        """运行 shell 命令"""
        body = {'command': command, 'agent': agent}
        if model:
            body['model'] = model
        return self._request('POST', f'/session/{session_id}/shell', json=body)

    # ========== 项目 ==========
    
    def list_projects(
        self, 
        directory: Optional[str] = None, 
        workspace: Optional[str] = None
    ) -> List[Dict]:
        """列出项目"""
        params = {}
        if directory:
            params['directory'] = directory
        if workspace:
            params['workspace'] = workspace
        return self._request('GET', '/project', params=params)

    def get_current_project(
        self, 
        directory: Optional[str] = None, 
        workspace: Optional[str] = None
    ) -> Dict:
        """获取当前项目"""
        params = {}
        if directory:
            params['directory'] = directory
        if workspace:
            params['workspace'] = workspace
        return self._request('GET', '/project/current', params=params)

    def init_git(
        self, 
        directory: Optional[str] = None, 
        workspace: Optional[str] = None
    ) -> Dict:
        """初始化 git 仓库"""
        params = {}
        if directory:
            params['directory'] = directory
        if workspace:
            params['workspace'] = workspace
        return self._request('POST', '/project/git/init', params=params)

    def update_project(self, project_id: str, updates: Dict) -> Dict:
        """更新项目"""
        return self._request('PATCH', f'/project/{project_id}', json=updates)

    # ========== 文件 ==========
    
    def list_files(
        self, 
        path: str = '.',
        directory: Optional[str] = None,
        workspace: Optional[str] = None,
    ) -> List[Dict]:
        """列出文件"""
        params = {'path': path}
        if directory:
            params['directory'] = directory
        if workspace:
            params['workspace'] = workspace
        return self._request('GET', '/file', params=params)

    def read_file(
        self, 
        path: str,
        directory: Optional[str] = None,
        workspace: Optional[str] = None,
    ) -> Dict:
        """读取文件内容"""
        params = {'path': path}
        if directory:
            params['directory'] = directory
        if workspace:
            params['workspace'] = workspace
        return self._request('GET', '/file/content', params=params)

    def get_file_status(self) -> List[Dict]:
        """获取文件状态"""
        return self._request('GET', '/file/status')

    # ========== 搜索 ==========
    
    def search_text(
        self, 
        pattern: str,
        directory: Optional[str] = None,
        workspace: Optional[str] = None,
    ) -> List[Dict]:
        """搜索文本"""
        params = {'pattern': pattern}
        if directory:
            params['directory'] = directory
        if workspace:
            params['workspace'] = workspace
        return self._request('GET', '/find', params=params)

    def find_files(
        self,
        query: str,
        file_type: Optional[str] = None,
        directory: Optional[str] = None,
        limit: Optional[int] = None,
    ) -> List[str]:
        """查找文件"""
        params = {'query': query}
        if file_type:
            params['type'] = file_type
        if directory:
            params['directory'] = directory
        if limit:
            params['limit'] = limit
        return self._request('GET', '/find/file', params=params)

    def find_symbols(self, query: str) -> List[Dict]:
        """查找符号"""
        return self._request('GET', '/find/symbol', params={'query': query})

    # ========== 配置 ==========
    
    def get_config(self) -> Dict:
        """获取配置"""
        return self._request('GET', '/config')

    def update_config(self, updates: Dict) -> Dict:
        """更新配置"""
        return self._request('PATCH', '/config', json=updates)

    def get_providers(self) -> Dict:
        """获取提供商和默认模型"""
        return self._request('GET', '/config/providers')

    def list_providers(self) -> Dict:
        """列出所有提供商"""
        return self._request('GET', '/provider')

    def get_provider_auth(self) -> Dict:
        """获取提供商认证方式"""
        return self._request('GET', '/provider/auth')

    def set_auth(self, provider_id: str, credentials: Dict) -> bool:
        """设置认证凭据"""
        return self._request('PUT', f'/auth/{provider_id}', json=credentials)

    def remove_auth(self, provider_id: str) -> bool:
        """移除认证凭据"""
        return self._request('DELETE', f'/auth/{provider_id}')

    # ========== 代理/命令/技能 ==========
    
    def list_agents(self) -> List[Dict]:
        """列出可用代理"""
        return self._request('GET', '/agent')

    def list_commands(self) -> List[Dict]:
        """列出可用命令"""
        return self._request('GET', '/command')

    def list_skills(self) -> List[Dict]:
        """列出可用技能"""
        return self._request('GET', '/skill')

    # ========== MCP ==========
    
    def get_mcp_status(self) -> Dict:
        """获取 MCP 服务器状态"""
        return self._request('GET', '/mcp')

    def add_mcp_server(self, name: str, config: Dict) -> Dict:
        """添加 MCP 服务器"""
        return self._request('POST', '/mcp', json={'name': name, 'config': config})

    def connect_mcp_server(self, name: str) -> Dict:
        """连接 MCP 服务器"""
        return self._request('POST', f'/mcp/{name}/connect')

    def disconnect_mcp_server(self, name: str) -> Dict:
        """断开 MCP 服务器"""
        return self._request('POST', f'/mcp/{name}/disconnect')

    # ========== 其他 ==========
    
    def get_vcs_info(self) -> Dict:
        """获取 VCS 信息"""
        return self._request('GET', '/vcs')

    def get_path_info(self) -> Dict:
        """获取路径信息"""
        return self._request('GET', '/path')

    def get_lsp_status(self) -> List[Dict]:
        """获取 LSP 状态"""
        return self._request('GET', '/lsp')

    def get_formatter_status(self) -> List[Dict]:
        """获取格式化器状态"""
        return self._request('GET', '/formatter')

    def list_permissions(self) -> List[Dict]:
        """列出权限请求"""
        return self._request('GET', '/permission')

    def respond_to_permission(
        self, 
        session_id: str, 
        permission_id: str, 
        response: str,
        remember: bool = False,
    ) -> bool:
        """响应权限请求"""
        return self._request('POST', f'/session/{session_id}/permissions/{permission_id}', json={
            'response': response,
            'remember': remember,
        })

    def list_pty(self) -> List[Dict]:
        """列出 PTY 会话"""
        return self._request('GET', '/pty')

    def create_pty(self) -> Dict:
        """创建 PTY 会话"""
        return self._request('POST', '/pty')


# 使用示例
if __name__ == '__main__':
    client = OpenCodeAPI(
        base_url=os.getenv('OPENCODE_SERVER_BASE_URL', 'http://127.0.0.1:4096'),
        username=os.getenv('OPENCODE_SERVER_USERNAME', 'opencode'),
        password=os.getenv('OPENCODE_SERVER_PASSWORD'),
    )
    
    # 健康检查
    print("健康检查:", client.health())
    
    # 列出会话
    print("会话列表:", client.list_sessions())
    
    # 创建会话
    session = client.create_session(title='测试会话')
    print("创建会话:", session)
    
    # 发送消息
    response = client.send_message(
        session['id'],
        parts=[{'type': 'text', 'text': '你好！'}]
    )
    print("消息响应:", response)
